core/src/main/java/org/apache/accumulo/core/rpc/clients/ManagerClient.java

```diff
@@ -36,30 +41,36 @@ public interface ManagerClient<C extends TServiceClient> {
   default C getManagerConnection(Logger log, ThriftClientTypes<C> type, ClientContext context) {
     checkArgument(context != null, "context is null");
 
-    List<String> locations = context.getManagerLocations();
+    ServiceLockPath slp = context.getServerPaths().getManager(true);
 
-    if (locations.isEmpty()) {
+    if (slp == null) {
       log.debug("No managers...");
       return null;
     }
 
-    HostAndPort manager = HostAndPort.fromString(locations.get(0));
-    if (manager.getPort() == 0) {
-      return null;
-    }
+    ZcStat stat = new ZcStat();
+    Optional<ServiceLockData> sld = ServiceLock.getLockData(context.getZooCache(), slp, stat);
```

Comment:
There are some codepaths that only use ServiceLockPaths to see if servers exist in ZooKeeper, without using the lock data. For those codepaths, this would cause extra load on ZK, time, and memory. We would have to do analysis to see if it's worth the single codepath.